src/configure/clients/modaTemplateServiceClient.ts

```diff
@@ -0,0 +1,87 @@
+
+import { RepositoryAnalysis } from "azureintegration-repoanalysis-client-internal";
+import { RestClient } from "typed-rest-client";
+import vscodeUri from "vscode-uri";
+import { ExtendedPipelineTemplate } from "../model/Contracts";
+import { StringMap } from "../model/models";
+import { TemplateInfo } from "../model/templateModels";
+import { ITemplateServiceClient } from "./ITemplateServiceClient";
+
+export class ModaTemplateServiceClient implements ITemplateServiceClient {
+    private restClient: RestClient;
+    private githubPat: string;
+    private pathUrl: string;
+    private readonly apiVersion = "6.0-preview.1";
+    private readonly extendedPipelineTemplateResource = "ExtendedPipelineTemplates";
+    private readonly templatesInfoResource = "TemplatesInfo";
+    private readonly templateAssetFilesResource = "TemplateAssetFiles";
+
+    constructor(url: string, githubPat: string) {
+        const u = vscodeUri.parse(url);
+        this.restClient = new RestClient("deploy-to-azure", u.scheme + "://" + u.authority);
+        this.pathUrl = u.path;
+        this.githubPat = githubPat;
+    }
+
+    public async getTemplates(body: RepositoryAnalysis): Promise<TemplateInfo[]> {
+        const requestUri = this.pathUrl + this.templatesInfoResource;
+        const requestOptions = {
+            acceptHeader: "application/json",
+            additionalHeaders: {
+                "Authorization": "Bearer " + this.githubPat
+            },
+            queryParameters: {
+                'api-version': this.apiVersion
+            }
+        };
+        return <TemplateInfo[]>(await this.restClient.create(requestUri, body, requestOptions)).result;
```

Comment:
I think this line is the major problem why you can't keep the code common. can't we change this restclient to have same interface for both types of APIs